examples/threading/threading.c

```diff
@@ -14,6 +18,57 @@ void* threadfunc(void* thread_param)
     // TODO: wait, obtain mutex, wait, release mutex as described by thread_data structure
     // hint: use a cast like the one below to obtain thread arguments from your parameter
     //struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+    
+    struct thread_data* thread_func_args = (struct thread_data*) thread_param;
+    
+    //we shall use the function nanosleep(), as sleep() itself is implemented in Linux using nanosleep()
+    //Reference can be found here: https://man7.org/linux/man-pages/man2/nanosleep.2.html
+    //Reference for code structure: https://linuxhint.com/nanosleep-function-c/
+    struct timespec ts_required = {0, (thread_func_args->wait_to_obtain_ms)*TIMESCALE_CONVERSION_FACTOR}; //multiplied by 10^6 to convert ms to ns
+    struct timespec ts_remaining;
+    
+    int return_status = nanosleep(&ts_required, &ts_remaining);
+    
+    while(return_status == -1){
+    	if(errno == EINTR){ //EINTR means the pause has been interrupted by signal. Reference at above link
+    		//sleep for remaining time
+    		ts_required.tv_sec = 0;
+    		ts_remaining.tv_nsec = ts_required.tv_nsec;
+    		return_status = nanosleep(&ts_required, &ts_remaining);
+    	}else //Error has occured in nanosleep()
+    	{
+    		ERROR_LOG("Sleep error. Exiting...");
+    		thread_func_args->thread_complete_success = false;
+    		return (void *)thread_func_args;
+    	}
+    }
```

Comment:
@prvbharadwaj1731 consider using a function for this rather than copying and pasting the while block